clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java

```diff
@@ -673,6 +673,11 @@ else if (value instanceof String)
                         return value;
                     } else if (value instanceof String) {
                         return Short.parseShort(trimmed);
+                    } else if (value instanceof Integer || value instanceof Long) {
+                        if (((Number) value).longValue() > Short.MAX_VALUE) {
+                            throw new ConfigException(name, value, "Expected value to be a 16-bit integer (short), but it was > " + Short.MAX_VALUE);
+                        }
+                        return ((Number) value).shortValue();
```

Comment:
Do we have to include this change in this PR? It'd be nice to keep this PR focused only on the changes proposed in KIP-465.